### PR TITLE
change str validation for pref type

### PR DESCRIPTION
**Pull request by tiftran**

### CODE REVIEW

File: app/experimenter/experiments/models.py [hunk 1]
```diff
@@ -545,7 +545,7 @@ class ExperimentVariant(models.Model):
     is_control = models.BooleanField(default=False)
     description = models.TextField(default="")
     ratio = models.PositiveIntegerField(default=1)
-    value = JSONField(blank=True, null=True)
+    value = models.CharField(max_length=10240, blank=False, null=True)
```
Comment (jaredkerim) [inline app/experimenter/experiments/models.py]: @tiftran Ah my bad I should've said TextField here, can you switch it?

File: app/experimenter/experiments/models.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 from django.conf import settings
 from django.contrib.auth import get_user_model
-from django.contrib.postgres.fields import JSONField
+
```
Comment (jaredkerim) [inline app/experimenter/experiments/models.py]: don't need an empty line here

File: app/experimenter/experiments/tests/test_forms.py [hunk 3]
```diff
@@ -993,21 +993,21 @@ def test_form_is_invalid_if_branches_have_duplicate_pref_values(self):
...
         form = ExperimentVariantsPrefForm(request=self.request, data=self.data)
         self.assertFalse(form.is_valid())
         self.assertIn("value", form.variants_formset.errors[0])
         self.assertIn("value", form.variants_formset.errors[1])
         self.assertNotIn("value", form.variants_formset.errors[2])
 
-    def test_form_is_invalid_if_pref_value_has_invalid_json(self):
-        self.data["variants-0-value"] = "abc"  # missing quotes
-
+    def test_form_is_valid_if_pref_type_is_string(self):
```
Comment (jaredkerim) [inline app/experimenter/experiments/tests/test_forms.py]: Can you add two more similar tests, one for int type and one for bool type just so we have our bases covered?

File: app/experimenter/experiments/forms.py [hunk 4]
```diff
@@ -515,22 +515,28 @@ def clean(self):
...
-            if not form.is_valid():
-                # This probably means that JSONField got invalid JSON.
-                continue
-            # Form field is string, but we check value type of parsed JSON.
-            found_type = type(json.loads(form.cleaned_data["value"]))
-            if found_type != expected_type:
-                form.add_error(
-                    "value", f"Unexpected value type (should be {pref_type})"
-                )
+        if pref_type in (Experiment.PREF_TYPE_BOOL, Experiment.PREF_TYPE_INT):
```
Comment (jaredkerim) [inline app/experimenter/experiments/forms.py]: 👍

File: app/experimenter/experiments/tests/test_forms.py (hunk 3 above)
Comment (tiftran) [reply app/experimenter/experiments/tests/test_forms.py]: will do

File: app/experimenter/experiments/models.py (hunk 1 above)
Comment (tiftran) [reply app/experimenter/experiments/models.py]: Would the max_length be needed it's a TextField?

File: app/experimenter/experiments/models.py [hunk 5]
```diff
@@ -545,7 +544,7 @@ class ExperimentVariant(models.Model):
     is_control = models.BooleanField(default=False)
     description = models.TextField(default="")
     ratio = models.PositiveIntegerField(default=1)
-    value = JSONField(blank=True, null=True)
+    value = models.TextField(max_length=10240, blank=False, null=True)
```
Comment (jaredkerim) [inline app/experimenter/experiments/models.py]: Sorry I forgot to mention, you don't need to specify the max_length here, the only thing it does is make the text field in the admin comically large 🐶 just take that out and regenerate the migration and we're good to go.

File: app/experimenter/experiments/models.py (hunk 1 above)
Comment (jaredkerim) [reply app/experimenter/experiments/models.py]: Oh sorry you asked here, I didn't see, no for TextField the only thing max_length does is set the size of the input field, so you usually just omit it and specify it on the actual widget (which you've done correctly already)

**Comment by jaredkerim** [comment]
@tiftran Perfect, looks ready to go, just rebase the migration on master and we'll land this today :)

**Pull request merged**